tests/Jira/Issues/WalkerTest.php

```diff
@@ -238,6 +238,33 @@ public function testIssuesPassedThroughDelegate()
 		);
 	}
 
+	public function testCounting()
+	{
+		// Full 1st page.
+		$search_response1 = $this->generateSearchResponse('PRJ1', 5, 7);
+		$this->api->search('test jql', 0, 5, 'description')->willReturn($search_response1);
+
+		// Incomplete 2nd page.
+		$search_response2 = $this->generateSearchResponse('PRJ2', 2, 7);
+		$this->api->search('test jql', 5, 5, 'description')->willReturn($search_response2);
+
+		$walker = $this->createWalker(5);
+		$walker->push('test jql', 'description');
+
+		$this->assertEquals(7, count($walker));
```

Comment:
Counting would involve querying 1st page of issues and the `foreach` after it ensures, that this query won't loose all issues on 1st page as a result.